### PR TITLE
WT-13092 Perform dirty eviction when the page has obsolete time window (5.0 backport) (#10874)

### PR DESCRIPTION
**Pull request by etienneptl**
Before evicting a clean page, mark the page dirty to perform a dirty eviction instead of clean page eviction when the page has obsolete time window.

---------

(cherry picked from commit 7a4dbfa352a895fdaf35f17d65575d833b4003f3)

### CODE REVIEW

**Comment by etienneptl** [comment]
I will set the new setting to 0 as the default value.

File: test/evergreen/find_cmake.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/sh
-set -o errexit  # Exit the script with error if any of the commands fail
+#set -o errexit  # Exit the script with error if any of the commands fail
```
Comment (kommiharibabu) [inline test/evergreen/find_cmake.sh:2]: same here as well?

File: dist/s_all [hunk 2]
```diff
@@ -83,7 +83,7 @@ run "python log.py"
 run "python stat.py"
 run "sh ./s_copyright"
 run "sh ./s_style"
-run "./s_clang-format"
+# run "./s_clang-format"
```
[the remark this answers is not in the document]
Comment (etienneptl) [reply dist/s_all:86]: Thank your for spotting those two, my bad.

**Comment by etienneptl** [comment]
The PR is missing the test somehow.

**Comment by etienneptl** [comment]
Facing multiple errors:
```
======================================================================
FAIL: test_eviction02.test_eviction02.test_evict(100) (subunit.RemotedTestCase)
test_eviction02.test_eviction02.test_evict(100)
----------------------------------------------------------------------
testtools.testresult.real._StringException: Traceback (most recent call last):
  File "/home/ubuntu/wiredtiger/test/suite/test_eviction02.py", line 102, in test_evict
    self.check_stat(prev_obsolete_tw_value, current_obsolete_tw_value)
  File "/home/ubuntu/wiredtiger/test/suite/test_eviction02.py", line 54, in check_stat
    assert diff <= threshold, f"Unexpected number of pages with obsolete tw cleaned: {diff} (max {threshold})"
AssertionError: Unexpected number of pages with obsolete tw cleaned: 201 (max 200)
```

And:
```
[2024/09/24 11:07:41.775] Contents from WT_TEST/test_eviction02.test_eviction02.test_evict_50/stderr.txt:
[2024/09/24 11:07:41.775] [1727138982:703193][6860:0x7f4c83268700], test_eviction02.test_eviction02.test_evict(50), file:eviction02.wt, eviction-server: __wt_ref_addr_copy, 1434: __wt_session_gen(session, WT_GEN_SPLIT) != 0
[2024/09/24 11:07:41.775] [1727138982:703276][6860:0x7f4c83268700], test_eviction02.test_eviction02.test_evict(50), file:eviction02.wt, eviction-server: __wt_abort, 28: aborting WiredTiger library
```

**Comment by etienneptl** [comment]
Closing as won't do for now.

**Pull request closed without merging**